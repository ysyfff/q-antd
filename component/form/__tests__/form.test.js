import React from 'react';
import enzyme from 'enzyme';
import * as mobx from 'mobx';
const { mount, render } = enzyme;
const { observable, toJS } = mobx;
import Form from '../index';
const { FormItem } = Form;
import Input_ from '../../input_';
import { observer } from 'mobx-react';

describe('测试Form', () => {
  const model = observable({
    name: '',
    pwd: '',
  });
  let jj = null;
  const rules = observable({
    name: [
      {
        required: true,
        message: '必须填写',
        trigger: 'blur',
      },
      // {
      //   validator(rule, value, callback) {
      //     jj.validate('pwdRule').then((res) => {

      //     })
      //   },
      //   trigger: 'change'
      // }
    ],
    pwdRule: [
      {
        required: false,
        message: '',
        trigger: 'change',
      },
      {
        validator(rule, value, callback) {
          callback('可以了')
        }
      }
    ],
  });


  @observer
  class Demo extends React.Component {
    render() {
      return (
        <Form model={model} rules={rules} ref={ref => jj = ref}>
          <FormItem label="姓名" prop="name">
            <Input_ duplex="name" />
          </FormItem>
          <FormItem label="密码" prop="pwdRule">
            <Input_ duplex="pwd" />
          </FormItem>
        </Form>
      );
    }
  }

  const wrapper = mount(<Demo />);
  it('姓名为必填，label应该有必填星标', () => {
    // jest.runAllTimers();
    // expect(wrapper).toMatchSnapshot();
    expect(
      wrapper
        .find('label')
        .at(0)
        .hasClass('ant-form-item-required')
    ).toBe(true);
    expect(
      wrapper
        .find('label')
        .at(0)
        .hasClass('i-form-label')
    ).toBe(true);
  });
  it('改变姓名时，校验密码规则', () => {
    console.log('wtfffffffffffff')
    wrapper.find('input').at(0).simulate('change', { target: { value: 1 } });
    // wrapper.find('input').at(0).simulate('change', { target: { value: 1 } });
    console.log(wrapper.find('input').at(0).prop('value'), 'Niiiiiiiii')
    
      expect(wrapper.find('.ant-form-explain')).toHaveLength(1);

    // setTimeout(()=>{
    // expect(wrapper.find('.ant-form-explain')).toHaveLength(1);
    // expect(wrapper.find('.ant-form-explain').innerText).toBe('可以了呢是')
    // }, 0);
  })

  it('姓名2为非必填，label不能有必填星标', () => {
    expect(
      wrapper
        .find('label')
        .at(1)
        .hasClass('ant-form-item-required')
    ).toBe(false);
  });

  
});


