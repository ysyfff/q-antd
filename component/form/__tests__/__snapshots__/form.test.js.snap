// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试Form 我来打个酱油 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "i-form",
          "style": "display:flex;align-items:flex-end",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "style": "flex:1",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "i-form-item-wrap",
                  "style": "display:flex",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "rules": "[object Object]",
                      "style": "display:flex;align-items:center;flex-direction:row",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "style": "display:flex",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "class": "i-form-label false false",
                              "style": "width:auto;text-align:right",
                              "title": "姓名",
                            },
                            "children": Array [
                              Object {
                                "data": "姓名",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "label",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "class": undefined,
                              "style": undefined,
                              "title": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "class": undefined,
                              "style": undefined,
                              "title": undefined,
                            },
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": Object {
                          "attribs": Object {
                            "class": "i-form-item",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {
                                "class": "i-form-item ",
                              },
                              "children": Array [
                                Object {
                                  "attribs": Object {
                                    "class": "ant-input",
                                    "duplexer": "",
                                    "type": "text",
                                    "value": "",
                                  },
                                  "children": Array [],
                                  "name": "input",
                                  "namespace": "http://www.w3.org/1999/xhtml",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "tag",
                                  "x-attribsNamespace": Object {
                                    "class": undefined,
                                    "duplexer": undefined,
                                    "type": undefined,
                                    "value": undefined,
                                  },
                                  "x-attribsPrefix": Object {
                                    "class": undefined,
                                    "duplexer": undefined,
                                    "type": undefined,
                                    "value": undefined,
                                  },
                                },
                              ],
                              "name": "div",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "class": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "class": undefined,
                              },
                            },
                          ],
                          "name": "div",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "style": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "style": undefined,
                        },
                      },
                      Object {
                        "attribs": Object {
                          "class": "i-form-item",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "class": "i-form-item ",
                            },
                            "children": Array [
                              Object {
                                "attribs": Object {
                                  "class": "ant-input",
                                  "duplexer": "",
                                  "type": "text",
                                  "value": "",
                                },
                                "children": Array [],
                                "name": "input",
                                "namespace": "http://www.w3.org/1999/xhtml",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "tag",
                                "x-attribsNamespace": Object {
                                  "class": undefined,
                                  "duplexer": undefined,
                                  "type": undefined,
                                  "value": undefined,
                                },
                                "x-attribsPrefix": Object {
                                  "class": undefined,
                                  "duplexer": undefined,
                                  "type": undefined,
                                  "value": undefined,
                                },
                              },
                            ],
                            "name": "div",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "class": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "class": undefined,
                            },
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {
                            "style": "display:flex",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {
                                "class": "i-form-label false false",
                                "style": "width:auto;text-align:right",
                                "title": "姓名",
                              },
                              "children": Array [
                                Object {
                                  "data": "姓名",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "text",
                                },
                              ],
                              "name": "label",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "class": undefined,
                                "style": undefined,
                                "title": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "class": undefined,
                                "style": undefined,
                                "title": undefined,
                              },
                            },
                          ],
                          "name": "div",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "style": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "style": undefined,
                          },
                        },
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "rules": undefined,
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "rules": undefined,
                      "style": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "style": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "style": "display:flex;align-items:flex-end;justify-content:flex-start",
              },
              "children": Array [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "style": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "style": undefined,
            },
          },
          Object {
            "attribs": Object {
              "style": "display:flex;align-items:flex-end;justify-content:flex-start",
            },
            "children": Array [],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "style": "flex:1",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "i-form-item-wrap",
                    "style": "display:flex",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "rules": "[object Object]",
                        "style": "display:flex;align-items:center;flex-direction:row",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "style": "display:flex",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {
                                "class": "i-form-label false false",
                                "style": "width:auto;text-align:right",
                                "title": "姓名",
                              },
                              "children": Array [
                                Object {
                                  "data": "姓名",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "text",
                                },
                              ],
                              "name": "label",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "class": undefined,
                                "style": undefined,
                                "title": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "class": undefined,
                                "style": undefined,
                                "title": undefined,
                              },
                            },
                          ],
                          "name": "div",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": Object {
                            "attribs": Object {
                              "class": "i-form-item",
                            },
                            "children": Array [
                              Object {
                                "attribs": Object {
                                  "class": "i-form-item ",
                                },
                                "children": Array [
                                  Object {
                                    "attribs": Object {
                                      "class": "ant-input",
                                      "duplexer": "",
                                      "type": "text",
                                      "value": "",
                                    },
                                    "children": Array [],
                                    "name": "input",
                                    "namespace": "http://www.w3.org/1999/xhtml",
                                    "next": null,
                                    "parent": [Circular],
                                    "prev": null,
                                    "type": "tag",
                                    "x-attribsNamespace": Object {
                                      "class": undefined,
                                      "duplexer": undefined,
                                      "type": undefined,
                                      "value": undefined,
                                    },
                                    "x-attribsPrefix": Object {
                                      "class": undefined,
                                      "duplexer": undefined,
                                      "type": undefined,
                                      "value": undefined,
                                    },
                                  },
                                ],
                                "name": "div",
                                "namespace": "http://www.w3.org/1999/xhtml",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "tag",
                                "x-attribsNamespace": Object {
                                  "class": undefined,
                                },
                                "x-attribsPrefix": Object {
                                  "class": undefined,
                                },
                              },
                            ],
                            "name": "div",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": [Circular],
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "class": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "class": undefined,
                            },
                          },
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "style": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "style": undefined,
                          },
                        },
                        Object {
                          "attribs": Object {
                            "class": "i-form-item",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {
                                "class": "i-form-item ",
                              },
                              "children": Array [
                                Object {
                                  "attribs": Object {
                                    "class": "ant-input",
                                    "duplexer": "",
                                    "type": "text",
                                    "value": "",
                                  },
                                  "children": Array [],
                                  "name": "input",
                                  "namespace": "http://www.w3.org/1999/xhtml",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "tag",
                                  "x-attribsNamespace": Object {
                                    "class": undefined,
                                    "duplexer": undefined,
                                    "type": undefined,
                                    "value": undefined,
                                  },
                                  "x-attribsPrefix": Object {
                                    "class": undefined,
                                    "duplexer": undefined,
                                    "type": undefined,
                                    "value": undefined,
                                  },
                                },
                              ],
                              "name": "div",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "class": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "class": undefined,
                              },
                            },
                          ],
                          "name": "div",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": Object {
                            "attribs": Object {
                              "style": "display:flex",
                            },
                            "children": Array [
                              Object {
                                "attribs": Object {
                                  "class": "i-form-label false false",
                                  "style": "width:auto;text-align:right",
                                  "title": "姓名",
                                },
                                "children": Array [
                                  Object {
                                    "data": "姓名",
                                    "next": null,
                                    "parent": [Circular],
                                    "prev": null,
                                    "type": "text",
                                  },
                                ],
                                "name": "label",
                                "namespace": "http://www.w3.org/1999/xhtml",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "tag",
                                "x-attribsNamespace": Object {
                                  "class": undefined,
                                  "style": undefined,
                                  "title": undefined,
                                },
                                "x-attribsPrefix": Object {
                                  "class": undefined,
                                  "style": undefined,
                                  "title": undefined,
                                },
                              },
                            ],
                            "name": "div",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": [Circular],
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "style": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "style": undefined,
                            },
                          },
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "rules": undefined,
                        "style": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "rules": undefined,
                        "style": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "style": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "style": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
